cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -528,6 +540,29 @@ public Builder setDateTime(String fieldName, @Nullable LocalDateTime localDateTi
       return this;
     }
 
+    /**
+     * Sets the java.nio.ByteBuffer for a Fixed datatype field.
+     *
+     * @param fieldName String field name
+     * @param byteBuffer {@link ByteBuffer} value , is nullable
+     * @return {@link Builder} for the StructuredRecord
+     */
+    public Builder setFixed(String fieldName, @Nullable ByteBuffer byteBuffer) {
+      Schema logicalSchema = validateAndGetLogicalTypeSchema(validateAndGetField(fieldName, byteBuffer),
+        EnumSet.of(LogicalType.FIXED));
+      if (byteBuffer == null) {
+        fields.put(fieldName, null);
+        return this;
+      }
+      if (byteBuffer.array().length !=  logicalSchema.getFixedSize()) {
```

Comment:
What if the ByteBuffer is not backed by array?

Reply:
This is a good point. ByteBuffers may not be supported by aa byte array. 

You can use the `remaining()` method to count how many bytes are remaining in the byte array, and this should match the fixed size.

Reply:
Changed to remaining